python/pyarrow/parquet.py

```diff
@@ -825,8 +1175,11 @@ def __init__(self, path_or_paths, filesystem=None, schema=None,
         if validate_schema:
             self.validate_schemas()
 
-        if filters:
+        if filters is not None:
```

Comment:
`filters` can now be either List[Tuple] or List[List[Tuple]] and can either filter on partitions or row groups depending on if the dataset is partitioned. Either way the current state should be mentioned in the docstring (unless that will be worked out before this PR is merged)